common/hyper_parameter.go

```diff
@@ -3,3 +3,8 @@ package common
 import "time"
 
 const RPC_TIMEOUT = 1000 * time.Millisecond
+const HEARTBEAT_INTERVAL = 3 * time.Second
+const HEARTBEAT_TIMEOUT = 6 * time.Second
+const HEARTBEAT_MONITOR_INTERVAL = 15 * time.Second
```

Comment:
This constant needs renaming or comments. Not 100% sure about what it means.

Reply:
By reading its reference I figured it out.

Reply:
Why not merge `HEARTBEAT_TIMEOUT` and `HEARTBEAT_MONITOR_INTERVAL`.
Say datanodes send HB every 3s. The namenode maintain a set (of datanodes). On receiving a HB, it adds the datanode to the set. Every time the HB monitor wakes up, just consider all datanodes in this set as alive, and then clear the set.

Reply:
Using `isAlive` field, as you did, is also an alternative to a set. Pay attention to two points:
1. avoiding getting timestamp, since syscalls are expensive;
2. adding another rwlock just for HB; these background tasks should be as non-disruptive as possible.